lib/tasks/data_key_validation.rake

```diff
@@ -0,0 +1,52 @@
+
+namespace :data_key_validation do
+  desc "Validates the value of ENV['CONJUR_DATA_KEY']"
+  # This task validate the master data key which is stored in `ENV['CONJUR_DATA_KEY']`
+  # Once the db is created, a secret will be created using the master key.
+  # Each time the server is restarted this task will attempt to retrieve the
+  # secret, if the provided key does not match the original master the secret retrieval
+  # will fail.
+  #
+  # What's the 'environment' task in Rake?
+  # the :validate => :environment means "do environment before validate".
+  # You can get access to your models, and in fact, your whole environment
+  # by making tasks dependent on the environment task.
+  task :validate => [:environment] do
+    data_key = ENV['CONJUR_DATA_KEY']
+
+    if data_key.nil? || data_key.empty?
+      raise Errors::System::EmptyConjurDataKey
+      exit 1
+    end
+
+    # try get the master key validator resource
+    resource_id = "!:!:data-key-validator-resource"
+
+    unless Resource[resource_id]
```

Comment:
if you `unless` and then `else` it is better to turn these around and have an `if` instead so it is more readable. so you can do:
```
if Resource[resource_id]
  # we are in a server restart - verify you can decrypt the secret
else
  # we are in the first run - create the secret